### PR TITLE
Regex feature implemented, fixed goto crash

### PR DESCRIPTION
**Pull request by soumyamahunt**
Implemented find and replace with regular expression.
Fixed goto crash on zero input.

### CODE REVIEW

File: src/Notepads/Controls/GoTo/GoToControl.xaml.cs [hunk 1]
```diff
@@ -122,7 +122,7 @@ private void GoToBar_BeforeTextChanging(TextBox sender, TextBoxBeforeTextChangin
                 NotificationCenter.Instance.PostNotification(_resourceLoader.GetString("GoTo_NotificationMsg_InputError_InvalidInput"), 1500);
                 args.Cancel = true;
             }
-            else if (line > _maxLine)
+            else if (line > _maxLine || line == 0)
```
Comment (JasonStein) [inline src/Notepads/Controls/GoTo/GoToControl.xaml.cs]: just make it <=0 to make sure we are 100% safe.

**Comment by JasonStein** [comment]
Can you briefly describe how it works (regex)? And what should we expect, plus some examples?

**Comment by soumyamahunt** [comment]
> Can you briefly describe how it works (regex)? And what should we expect, plus some examples?

In a document open the find control and type your regular expression pattern in the find box.
Select the "Use Regular Expression" in the search options.
Optionally you can select or deselect the "Match Case" to whether take in account or ignore the cases in text respectively.
You can optionally replace one by one or all the words that match the Regular Expression pattern.

For example, open the readme file and give regular pattern "github/[a-z]+" and choose the "Use Regular Expression" feature (keep the "Match Case" unselected). This will give words that match that pattern (in this case user profiles). If you type the pattern "github/[A-Z]+" this will give same result if you keep "Match Case" unselected. If you select match case this would give different result. Alternativly patterns like "\b[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}\b" searches for all emails in a document.

**Comment by JasonStein** [comment]
LGTM, please help test as well.

**Pull request merged**